src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/scheduled/SubjectAccessRequestProcessor.kt

```diff
@@ -0,0 +1,139 @@
+package uk.gov.justice.digital.hmpps.subjectaccessrequestworker.scheduled
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.apache.commons.lang3.time.StopWatch
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.alerting.AlertsService
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.config.trackSarEvent
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.exception.SubjectAccessRequestException
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.models.Status
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.models.SubjectAccessRequest
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.services.ReportService
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.services.SubjectAccessRequestService
+import java.util.concurrent.TimeUnit
+
+@Component
+class SubjectAccessRequestProcessor(
+  private val subjectAccessRequestService: SubjectAccessRequestService,
+  private val reportService: ReportService,
+  private val alertsService: AlertsService,
+  private val telemetryClient: TelemetryClient,
+) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+    const val TIME_ELAPSED_KEY = "totalTimeElapsed"
+  }
+
+  @Scheduled(
+    fixedDelayString = "\${scheduled.subject-access-request-processor.interval-seconds:30}",
+    timeUnit = TimeUnit.SECONDS,
+    initialDelayString = "\${scheduled.subject-access-request-processor.initial-delay-seconds:30}",
+  )
+  fun execute() {
+    log.info("checking for available subject access requests to process")
+
+    var subjectAccessRequest: SubjectAccessRequest? = null
+    val stopWatch: StopWatch = StopWatch.create()
+
+    try {
+      subjectAccessRequest = getNextRequestOrNull()
+
+      subjectAccessRequest?.let {
+        processRequest(it, stopWatch)
+      } ?: run {
+        log.info("no available subject access requests to process")
+      }
+    } catch (ex: Exception) {
+      handleError(stopWatch, subjectAccessRequest, ex)
+    }
+  }
+
+  private fun getNextRequestOrNull(): SubjectAccessRequest? = subjectAccessRequestService
+    .findUnclaimed()
+    .takeIf { it.isNotEmpty() }
+    ?.let { it[0] }
```

Comment:
You could use firstOrNull here:
```suggestion
    ?.firstOrNull()
```